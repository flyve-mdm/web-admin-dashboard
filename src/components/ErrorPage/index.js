/*
 *   Copyright © 2018 Teclib. All rights reserved.
 *
 *   This file is part of web-mdm-dashboard
 *
 * web-mdm-dashboard is a subproject of Flyve MDM. Flyve MDM is a mobile
 * device management software.
 *
 * Flyve MDM is free software: you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 3
 * of the License, or (at your option) any later version.
 *
 * Flyve MDM is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * ------------------------------------------------------------------------------
 * @author     Gianfranco Manganiello (gmanganiello@teclib.com)
 * @author     Hector Rondon (hrondon@teclib.com)
 * @copyright  Copyright © 2018 Teclib. All rights reserved.
 * @license    GPLv3 https://www.gnu.org/licenses/gpl-3.0.html
 * @link       https://github.com/flyve-mdm/web-mdm-dashboard
 * @link       http://flyve.org/web-mdm-dashboard
 * @link       https://flyve-mdm.com
 * ------------------------------------------------------------------------------
 */

import React from "react"
import history from "../../shared/history"
import getQueryString from "../../shared/getQueryString"
import I18n from '../../shared/i18n'
import EmptyMessage from '../../components/EmptyMessage'

/**
 * Component with the display of the error pages
 * @function ErrorPage
 * @return {component}
 */
export default () => {
  /** Get the error type of the query string 'code' (by default '404'). */
  let errorCode = getQueryString(history).code
  if (errorCode !== '400' && errorCode !== '401' && errorCode !== '403' && errorCode !== '404' && errorCode !== '500') {
    errorCode = '404'
  }
  const title = I18n.t(`error.${errorCode}.title`)
  const message = I18n.t(`error.${errorCode}.message`)

  if (errorCode !== 404 && errorCode < 500) {
    return(
      <div
        className="empty-message"
        style={{width: '100%'}}
      >
        <div style={{width: '320px', margin: '0 auto'}}>
          <figure>
            <img
              alt="Flyve MDM Dashboard"
              src={require('../../assets/images/dashboard.svg')}
              style={{
                width: '80px',
              }}
            />
          </figure>
          <h1>{title}</h1>
          <p>{message}</p>
        </div>
      </div>
    )
  }

  return (
    <div className="authentication error" style={{ textAlign: 'center'}} >
      <section>
        <figure>
          <img
            alt="Flyve MDM Dashboard"
            src={require('../../assets/images/dashboard.svg')}
          />
        </figure>
        <h1>{I18n.t('commons.error')} {errorCode}</h1>
        <h3>{title}</h3>
        <p>{message}</p>
      </section>
      <footer>
        <a href="https://flyve-mdm.com/privacy-policy/">
          {I18n.t('commons.terms_and_conditions')}
        </a>
        <br />

        <span>
          © 2017 - 2018 Teclib'.
        </span>
        <br />
      </footer>
    </div>
  )
}
