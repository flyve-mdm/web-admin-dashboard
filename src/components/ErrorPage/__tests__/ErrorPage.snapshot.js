/*
 *   Copyright © 2018 Teclib. All rights reserved.
 *
 *   This file is part of web-mdm-dashboard
 *
 * web-mdm-dashboard is a subproject of Flyve MDM. Flyve MDM is a mobile
 * device management software.
 *
 * Flyve MDM is free software: you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 3
 * of the License, or (at your option) any later version.
 *
 * Flyve MDM is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * ------------------------------------------------------------------------------
 * @author     Gianfranco Manganiello (gmanganiello@teclib.com)
 * @author     Hector Rondon (hrondon@teclib.com)
 * @copyright  Copyright © 2018 Teclib. All rights reserved.
 * @license    GPLv3 https://www.gnu.org/licenses/gpl-3.0.html
 * @link       https://github.com/flyve-mdm/web-mdm-dashboard
 * @link       http://flyve.org/web-mdm-dashboard
 * @link       https://flyve-mdm.com
 * ------------------------------------------------------------------------------
 */

import React from 'react'
import renderer from 'react-test-renderer'

import ErrorPage from "../index"

describe('ErrorPage', () => {
  test('renders ErrorPage', () => {
    const component = renderer.create(
      <ErrorPage />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
