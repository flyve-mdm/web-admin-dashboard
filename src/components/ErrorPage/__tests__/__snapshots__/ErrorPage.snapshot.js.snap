// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage renders ErrorPage 1`] = `
<div
  className="empty-message"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "margin": "0 auto",
        "width": "320px",
      }
    }
  >
    <figure>
      <img
        alt="Flyve MDM Dashboard"
        src="dashboard.svg"
        style={
          Object {
            "width": "80px",
          }
        }
      />
    </figure>
    <h1>
      Oops! Page not Found.
    </h1>
    <p>
      We could not find the page you were looking for. Meanwhile, you may return to the dashboard.
    </p>
  </div>
</div>
`;
